### PR TITLE
5 commits: around yaml refactoring

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: mono_repo/lib/src/root_config.dart [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart' as y;
+
+import 'package_config.dart';
+import 'user_exception.dart';
+import 'utils.dart';
+
+class RootConfig extends MapBase<String, PackageConfig> {
```
Comment (jakemac53) [inline mono_repo/lib/src/root_config.dart]: Does this not have a consistent structure? (why not json_serializable?)

File: mono_repo/lib/src/yaml.dart [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  }
+}
+
+/// Strings that have special meaning in Yaml
+///
+/// Note: "yes" and "no" are not treated as `true`/`false` in pkg:yaml, but
+/// they are in many other parsers.
+const _yamlSpecialStrings = ['true', 'false', 'null', '...', 'yes', 'no'];
+
+final _yamlIndicators = [
```
Comment (jakemac53) [inline mono_repo/lib/src/yaml.dart]: Where are these coming from, is there a canonical list somewhere you can link to? Same for _yamlSpecialStrings.

File: mono_repo/test/yaml_write_test.dart [hunk 3]
```diff
@@ -0,0 +1,695 @@
...
+
+  test(testTitle, () {
+    var output = toYaml(source);
+    printOnFailure(['# start yaml', output, '# end yaml'].join('\n'));
+    var yaml = y.loadYaml(output);
+    expect(yaml, source);
+  });
+
+  if (source is! String) {
+    _testRoundTrip(source.toString());
```
Comment (jakemac53) [inline mono_repo/test/yaml_write_test.dart]: what is the purpose of this recursive call? can you make it more explicit?

File: mono_repo/test/yaml_write_test.dart [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+    }
+  });
+
+  group('naughty strings', () {
+    for (var source in _stringEscapeSamples.followedBy(_naughtyStrings)) {
+      _testRoundTrip(source);
+    }
+  });
+
+  for (var item in [
```
Comment (jakemac53) [inline mono_repo/test/yaml_write_test.dart]: lets move this list into a top level variable or something, its pretty hard to find the actual block for the loop, especially with the listed maps :joy_cat:

File: mono_repo/test/yaml_write_test.dart [hunk 5]
```diff
@@ -0,0 +1,695 @@
...
+  'Ⱦ',
+  'ヽ༼ຈل͜ຈ༽ﾉ ヽ༼ຈل͜ຈ༽ﾉ ',
+  '(｡◕ ∀ ◕｡)',
+  '｀ｨ(´∀｀∩',
+  '__ﾛ(,_,*)',
+  '・(￣∀￣)・:*:',
+  'ﾟ･✿ヾ╲(｡◕‿◕｡)╱✿･ﾟ',
+  ',。・:*:・゜’( ☻ ω ☻ )。・:*:・゜’',
+  '(╯°□°）╯︵ ┻━┻)',
+  '(ﾉಥ益ಥ）ﾉ﻿ ┻━┻',
```
Comment (jakemac53) [inline mono_repo/test/yaml_write_test.dart]: thats a new one lol

File: mono_repo/lib/src/root_config.dart (hunk 1 above)
Comment (kevmoo) [reply mono_repo/lib/src/root_config.dart]: This is a synthetic data structure composing `mono_config.yaml` + all of the `pkg_config.yaml` files. So it's not like we're parsing a single input file..

File: mono_repo/lib/src/yaml.dart (hunk 2 above)
Comment (kevmoo) [reply mono_repo/lib/src/yaml.dart]: ack

File: mono_repo/lib/src/root_config.dart (hunk 1 above)
Comment (jakemac53) [reply mono_repo/lib/src/root_config.dart]: hmm, name is super misleading then imo, PackageConfigs or something? It looks like its literally just the package configs right now right not the mono_config as well?

File: mono_repo/test/yaml_write_test.dart (hunk 3 above)
Comment (kevmoo) [reply mono_repo/test/yaml_write_test.dart]: ack

File: mono_repo/test/yaml_write_test.dart (hunk 5 above)
Comment (kevmoo) [reply mono_repo/test/yaml_write_test.dart]: uh huh

**Pull request merged**